addon/-private/class/modifier.ts

```diff
@@ -261,4 +295,71 @@ export default class ClassBasedModifier<S = DefaultSignature> {
   }
 }
 
+// We apply these here, against the prototype, so that there is only one of
+// these, rather than one per instance. We also only issue the deprecation once
+// per class for each of `args` and `element`.
+Object.defineProperty(ClassBasedModifier.prototype, 'args', {
+  enumerable: true,
+  get(this: InternalClassBasedModifier<unknown>) {
+    deprecate(
+      `ember-modifier (in ${this.constructor.name} at ${
+        new Error().stack
+      }): using \`this.args\` is deprecated. Access positional and named arguments directly in the \`modify\` hook instead.`,
+      !SEEN_CLASSES_FOR_ARGS.has(this.constructor),
+      {
+        id: 'ember-modifier.no-args-property',
+        for: 'ember-modifier',
+        since: {
+          available: '3.2.0',
+          enabled: '3.2.0',
+        },
+        until: '4.0.0',
+      }
+    );
+
+    if (DEBUG && !SEEN_CLASSES_FOR_ARGS.has(this.constructor)) {
+      SEEN_CLASSES_FOR_ARGS.add(this.constructor);
+    }
+
+    return this[Args];
+  },
+});
+
+Object.defineProperty(ClassBasedModifier.prototype, 'element', {
+  enumerable: true,
+  get(this: InternalClassBasedModifier<unknown>) {
+    deprecate(
+      `ember-modifier (in ${this.constructor.name} at ${
+        new Error().stack
+      }): using \`this.element\` is deprecated. Access the \`element\` as an argument in the \`modify\` hook instead.`,
+      !SEEN_CLASSES_FOR_ELEMENTS.has(this.constructor),
```

Comment:
Same thing as above, I think if we've issued the lifecycle deprecation we shouldn't also issue this one